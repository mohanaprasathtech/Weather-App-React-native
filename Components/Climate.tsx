/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
interface Props {
  fetchData: any;
}
const Climate: React.FC<Props> = props => {
  var bg: string;
  var message: string;
  var url: string;
  if (props.fetchData?.humidity > 80 && props.fetchData?.humidity < 130) {
    bg = '#344f70';
    message = 'Rainy Climate';
    url =
      'https://cdn1.iconfinder.com/data/icons/flat-mobile-app-icons/128/weather-512.png';
  } else if (
    props.fetchData?.humidity <= 80 &&
    props.fetchData?.humidity >= 50
  ) {
    bg = '#54b3d1';
    message = 'Chill Climate';
    url = 'https://icon-library.com/images/cooling-icon/cooling-icon-14.jpg';
  } else if (
    props.fetchData?.humidity <= 45 &&
    props.fetchData?.humidity >= 10
  ) {
    bg = '#f4c679';
    message = 'Very Hot';
    url =
      'https://png.pngtree.com/element_our/20190529/ourlarge/pngtree-weather-forecast-image_1197280.jpg';
  } else {
    bg = '#f4c679';
    message = 'Normal';
    url =
      'https://png.pngtree.com/element_our/20190529/ourlarge/pngtree-weather-forecast-image_1197280.jpg';
  }

  return (
    <View style={[styles.root, {backgroundColor: bg}]}>
      <View style={styles.container}>
        <Text style={styles.date}>{new Date().toDateString()}</Text>
        <Text
          style={
            styles.cityCountryText
          }>{`${props.fetchData?.location?.name}, ${props.fetchData?.location?.country}`}</Text>
        {props.fetchData?.humidity > 80 && props.fetchData?.humidity < 130 ? (
          <Image source={{uri: url}} style={styles.image} />
        ) : props.fetchData?.humidity <= 80 &&
          props.fetchData?.humidity >= 50 ? (
          <Image
            source={{
              uri: url,
            }}
            style={styles.image}
          />
        ) : (
          <Image
            source={{
              uri: url,
            }}
            style={styles.image}
          />
        )}
        <Text style={styles.temp}>{`${Math.round(
          props.fetchData?.current?.temp_c,
        )} °C`}</Text>
        <Text style={styles.cityCountryText}>{message}</Text>
        <Text style={styles.minmax}>{`Min ${Math.round(
          props.fetchData?.current?.temp_c,
        )} °C / Max ${Math.round(props.fetchData?.current?.temp_f)} °C`}</Text>
      </View>
      <View style={styles.row1}>
        <View>
          <Text style={styles.titletext}>Wind Speed</Text>
          <Text style={styles.detailtext}>
            {`${props.fetchData?.current?.wind_kph} km/h`}
          </Text>
        </View>
        <View style={{marginLeft: 30}}>
          <Text style={styles.titletext}>Precipitation</Text>
          <Text style={styles.detailtext}>
            {props.fetchData?.current?.precip_mm}
          </Text>
        </View>
      </View>
    </View>
  );
};
const styles = StyleSheet.create({
  root: {height: '100%'},
  container: {
    marginLeft: 'auto',
    marginRight: 'auto',
  },
  image: {
    height: 200,
    width: 200,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 20,
  },
  cityCountryText: {
    fontSize: 25,
    fontWeight: '400',
    color: 'white',
    textAlign: 'center',
  },
  date: {
    fontSize: 22,
    marginVertical: 10,
    color: 'white',
    textAlign: 'center',
    marginTop: 4,
  },
  temp: {
    fontSize: 38,
    color: 'white',
    textAlign: 'center',
    marginTop: 8,
  },
  minmax: {
    fontSize: 22,
    marginVertical: 10,
    fontWeight: '500',
    color: 'white',
    textAlign: 'center',
  },
  row1: {
    flexDirection: 'row',
    marginTop: 10,
    justifyContent: 'space-around',
    marginLeft: -18,
  },
  detailtext: {
    color: 'white',
    fontSize: 20,
    textAlign: 'center',
  },
  titletext: {
    fontSize: 19,
    color: 'white',
  },
});
export default Climate;
