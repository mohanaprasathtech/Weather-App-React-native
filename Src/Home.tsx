import React, {useState} from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity,
  TextInput,
  Image,
  Alert,
} from 'react-native';
import axios from 'axios';

interface Props {
  fetchData: any;
  setFetchData: any;
  navigation: any;
}
export async function handlefetch(final_link: string) {
  try {
    var res: any = await axios.get(final_link);

    return res.data;
  } catch (error) {
    Alert.alert('Enter Valid Capital name');
  }
}

const Demo: React.FC<Props> = props => {
  const [input, setInput] = useState<string>('');
  async function handleclick() {
    setInput('');
    var url = `https://api.openweathermap.org/data/2.5/weather?q=${input}&units=metric&appid=c8922c69c948f2e2b4bf08587bb7e7f0`;
    var finaldata: any = await handlefetch(url, props.navigation);
    props.setFetchData(finaldata);
    props.navigation.push('Info');
  }

  return (
    <View style={styles.container}>
      <Image
        source={{
          uri: 'https://icon-library.com/images/mostly-cloudy-icon/mostly-cloudy-icon-4.jpg',
        }}
        style={styles.logo}
      />
      <View style={styles.inputtext}>
        <TextInput
          placeholder="Enter Captial name"
          placeholderTextColor={'black'}
          value={input}
          onChangeText={val => setInput(val)}
          onSubmitEditing={handleclick}
        />
      </View>
      <View>
        <TouchableOpacity style={styles.btn} onPress={handleclick}>
          <Text style={styles.btntext}>Get Info</Text>
        </TouchableOpacity>
      </View>
      <Text style={styles.logotext}>All Weather Infomartion</Text>
      <Text style={[styles.logotext, {marginTop: 2}]}>of Location</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  logotext: {
    color: 'black',
    fontSize: 28,
    fontWeight: '500',
    textAlign: 'center',
    marginTop: 80,
  },
  logo: {
    height: 250,
    width: 300,
    marginLeft: 'auto',
    marginRight: 'auto',
    marginTop: 15,
  },
  container: {
    backgroundColor: '#13456d',
    flex: 1,
    height: '100%',
  },
  btn: {
    width: 100,
    backgroundColor: '#cabaa9',
    marginTop: 18,
    marginLeft: 'auto',
    marginRight: 'auto',
    height: 45,
    borderRadius: 18,
    alignItems: 'center',
    textAlign: 'center',
  },
  btntext: {
    color: 'black',
    marginTop: 13,
    fontSize: 15,
  },
  inputtext: {
    backgroundColor: '#cabaa9',
    color: 'white',
    borderRadius: 20,
    marginTop: 40,
    width: '90%',
    marginLeft: 'auto',
    marginRight: 'auto',
    alignItems: 'center',
  },
});

export default Demo;
