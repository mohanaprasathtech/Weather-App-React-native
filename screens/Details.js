import React from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  Text,
  useColorScheme,
  View,
} from 'react-native';
function Details({fetchdata}) {
  return (
    <View>
      <Text>{fetchdata?.name}</Text>
    </View>
  );
}

export default Details;
