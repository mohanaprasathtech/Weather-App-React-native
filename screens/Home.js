import axios from 'axios';
import React, {useEffect, useState} from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  StyleSheet,
  ImageBackground,
  Text,
  View,
  TouchableOpacity,
  TextInput,
} from 'react-native';

function Home({navigation, setFetchdata, fetchdata}) {
  const api = {
    key: 'c8922c69c948f2e2b4bf08587bb7e7f0',
  };

  const [input, setInput] = useState('');
  const [data, setData] = useState([]);

  function handleclick() {
    alert(input);
    setInput('');
    axios
      .get(
        `https://api.openweathermap.org/data/2.5/weather?q=${input}&units=metric&appid=${api.key}`,
      )
      .then(res => {
        //console.log(res.data);
        //setData(res.data);
        setFetchdata(res.data);
        navigation.navigate('Details');
      })
      .catch(err => console.log(err.message));
  }
  console.log(fetchdata);
  //console.log(data);
  return (
    <View style={styles.container}>
      <View style={styles.inputtext}>
        <TextInput
          placeholder="Enter city"
          placeholderTextColor={'black'}
          value={input}
          onChangeText={val => setInput(val)}
        />
      </View>
      <View>
        <TouchableOpacity style={styles.btn} onPress={handleclick}>
          <Text
            style={{
              color: 'black',
              marginTop: 3,
            }}>
            Get
          </Text>
        </TouchableOpacity>
      </View>
    </View>
  );
}
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'grey',
  },
  btn: {
    width: 100,
    backgroundColor: 'white',
    marginTop: 30,
    marginLeft: 'auto',
    marginRight: 'auto',
    height: 30,
    borderRadius: 20,
    alignItems: 'center',
    textAlign: 'center',
  },
  inputtext: {
    backgroundColor: 'white',
    marginTop: 150,
    borderRadius: 20,
    width: '90%',
    marginLeft: 'auto',
    marginRight: 'auto',
    alignItems: 'center',
  },
});
export default Home;
