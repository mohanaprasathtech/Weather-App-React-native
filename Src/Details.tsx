import React from 'react';
import {View} from 'react-native';
import Climate from '../Components/Climate';

interface Props {
  fetchData: any;
  navigation: any;
}
const Details: React.FC<Props> = props => {
  return (
    <View>
      {props.fetchData ? <Climate fetchData={props.fetchData} /> : <View />}
    </View>
  );
};
export default Details;
