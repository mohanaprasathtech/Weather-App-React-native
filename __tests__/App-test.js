/**
 * @format
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Climate from '../Components/Climate';
import Details from '../Src/Details';
import Home, {handlefetch} from '../Src/Home';

test('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
  const tree2 = renderer.create(<Climate />).toJSON();
  expect(tree2).toMatchSnapshot();
  const tree3 = renderer.create(<Details />).toJSON();
  expect(tree3).toMatchSnapshot();
});

const axios = require('axios');

jest.mock('axios');
it('API call test', async () => {
  axios.get.mockResolvedValue({
    data: {
      base: 'stations',
      clouds: {all: 89},
      cod: 200,
      coord: {lat: 10.95, lon: 78.0833},
      dt: 1661442234,
      id: 1267648,
      main: {
        feels_like: 26.17,
        grnd_level: 994,
        humidity: 78,
        pressure: 1007,
        sea_level: 1007,
        temp: 26.17,
        temp_max: 26.17,
        temp_min: 26.17,
      },
      name: 'Karur',
      sys: {country: 'IN', sunrise: 1661387888, sunset: 1661432522},
      timezone: 19800,
      visibility: 10000,
      weather: [
        {description: 'overcast clouds', icon: '04n', id: 804, main: 'Clouds'},
      ],
      wind: {deg: 279, gust: 8.28, speed: 4.71},
    },
  });
  const title = await handlefetch();
  expect(title.name).toEqual('Karur');
});
