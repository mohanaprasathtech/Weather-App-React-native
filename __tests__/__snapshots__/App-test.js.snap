// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#cabaa9",
        "borderRadius": 20,
        "color": "white",
        "marginLeft": "auto",
        "marginRight": "auto",
        "marginTop": 10,
        "width": "90%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Captial name"
      placeholderTextColor="black"
      value=""
    />
  </View>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#cabaa9",
          "borderRadius": 18,
          "height": 45,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 10,
          "opacity": 1,
          "textAlign": "center",
          "width": 100,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 15,
            "marginTop": 13,
          }
        }
      >
        Get Info
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#13456d",
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "height": "100%",
        "marginTop": 5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://icon-library.com/images/mostly-cloudy-icon/mostly-cloudy-icon-4.jpg",
        }
      }
      style={
        Object {
          "height": 250,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 100,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 28,
          "fontWeight": "500",
          "textAlign": "center",
        }
      }
    >
      All Weather Info of Location
    </Text>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Array [
      Object {
        "height": "100%",
      },
      Object {
        "backgroundColor": "#f4c679",
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": "auto",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "marginTop": 45,
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Thu Aug 25 2022
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      undefined, undefined
    </Text>
    <Image
      source={
        Object {
          "uri": "https://png.pngtree.com/element_our/20190529/ourlarge/pngtree-weather-forecast-image_1197280.jpg",
        }
      }
      style={
        Object {
          "height": 300,
          "marginLeft": "auto",
          "marginRight": "auto",
          "marginTop": 30,
          "width": 300,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 38,
          "marginTop": -10,
          "textAlign": "center",
        }
      }
    >
      NaN °C
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 25,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      Very Hot
    </Text>
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "fontWeight": "500",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      Min NaN °C / Max NaN °C
    </Text>
  </View>
</View>
`;

exports[`renders correctly 3`] = `
<View>
  <View />
</View>
`;
